File: src/main/java/MvvmDialog.java
import javax.swing.*;
import javax.swing.event.DocumentEvent;
import javax.swing.event.DocumentListener;
import java.awt.*;
import java.awt.event.*;

public class MvvmDialog extends JDialog {
    private JPanel contentPane;
    private JButton buttonOK;
    private JButton buttonCancel;
    private JTextField nameField;
    private JTextField descField;
    private JTextField activityNameField;
    private JTextField viewModelNameField;
    private JTextField layoutNameField;
    private JTextField packageField;
    private JLabel nameLabel;
    private JLabel descLabel;
    private JLabel activityNameLabel;
    private JLabel layoutNameLabel;
    private JLabel viewModelNameLabel;
    private JLabel packageLabel;
    private JRadioButton activityRadioButton;
    private JRadioButton fragmentRadioButton;
    private String modeName;

    public MvvmDialog(String packageName, String modeName, DialogCallBack dialogCallBack) {
        this.modeName = modeName;
        setContentPane(contentPane);
        setModal(true);
        getRootPane().setDefaultButton(buttonOK);
        initView(packageName);

        buttonOK.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                InputContentData data = new InputContentData();
                data.isActivity = activityRadioButton.isSelected();
                data.name = nameField.getText().trim();
                data.desc = descField.getText().trim();
                data.view = activityNameField.getText().trim();
                data.viewMode = viewModelNameField.getText().trim();
                data.layout = layoutNameField.getText().trim();
                data.packge = packageField.getText().trim();
                dialogCallBack.ok(data);
                dispose();
            }
        });

        buttonCancel.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                dispose();
            }
        });

        // call onCancel() when cross is clicked
        setDefaultCloseOperation(DO_NOTHING_ON_CLOSE);
        addWindowListener(new WindowAdapter() {
            public void windowClosing(WindowEvent e) {
                dispose();
            }
        });

        // call onCancel() on ESCAPE
        contentPane.registerKeyboardAction(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                dispose();
            }
        }, KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0), JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT);
    }

    private void initView(String packageName) {


        packageField.setText(packageName);
        //定义按钮组
        ButtonGroup group = new ButtonGroup();
        group.add(activityRadioButton);
        group.add(fragmentRadioButton);
        activityRadioButton.addChangeListener(changeEvent -> {
            onNameChange();
        });
        fragmentRadioButton.addChangeListener(changeEvent -> {
            onNameChange();
        });

        //焦点改变
        nameField.addFocusListener(new MyFocusListener());
        descField.addFocusListener(new MyFocusListener());
        activityNameField.addFocusListener(new MyFocusListener());
        viewModelNameField.addFocusListener(new MyFocusListener());
        layoutNameField.addFocusListener(new MyFocusListener());
        packageField.addFocusListener(new MyFocusListener());

        nameField.getDocument().addDocumentListener(new DocumentListener() {
            @Override
            public void insertUpdate(DocumentEvent documentEvent) {
                onNameChange();
            }

            @Override
            public void removeUpdate(DocumentEvent documentEvent) {
                onNameChange();
            }

            @Override
            public void changedUpdate(DocumentEvent documentEvent) {
                onNameChange();
            }
        });
    }

    public void onNameChange() {
        String modeNameSux = modeName.isEmpty() ? "" : modeName + "_";
        String nameText = nameField.getText().trim();
        if (!nameText.isEmpty()) {
            if (nameText.toLowerCase().contains("activity")) {
                nameText = nameText.toLowerCase().replace("activity", "");
            }
            if (nameText.toLowerCase().contains("viewmodel")) {
                nameText = nameText.toLowerCase().replace("viewmodel", "");
            }
            viewModelNameField.setText(nameText + "ViewModel");


            if (activityRadioButton.isSelected()) {
                activityNameField.setText(nameText + "Activity");

                layoutNameField.setText(modeNameSux + "activity" + Utils.activityToLayout(nameText));
            } else {
                activityNameField.setText(nameText + "Fragment");
                layoutNameField.setText(modeNameSux + "fragment" + Utils.activityToLayout(nameText));
            }
        } else {
            activityNameField.setText("");
            viewModelNameField.setText("");
            layoutNameField.setText("");
        }
    }

    private class MyFocusListener implements FocusListener {
        public void select(FocusEvent focusEvent, boolean isSelect) {
            nameLabel.setForeground(Color.decode("0x999999"));
            descLabel.setForeground(Color.decode("0x999999"));
            activityNameLabel.setForeground(Color.decode("0x999999"));
            layoutNameLabel.setForeground(Color.decode("0x999999"));
            viewModelNameLabel.setForeground(Color.decode("0x999999"));
            packageLabel.setForeground(Color.decode("0x999999"));
            if (focusEvent.getComponent() == nameField) {
                nameLabel.setForeground(isSelect ? Color.decode("0xff0000") : Color.decode("0x999999"));
            }
            if (focusEvent.getComponent() == descField) {
                descLabel.setForeground(isSelect ? Color.decode("0xff0000") : Color.decode("0x999999"));
            }
            if (focusEvent.getComponent() == activityNameField) {
                activityNameLabel.setForeground(isSelect ? Color.decode("0xff0000") : Color.decode("0x999999"));
            }
            if (focusEvent.getComponent() == layoutNameField) {
                layoutNameLabel.setForeground(isSelect ? Color.decode("0xff0000") : Color.decode("0x999999"));
            }
            if (focusEvent.getComponent() == viewModelNameField) {
                viewModelNameLabel.setForeground(isSelect ? Color.decode("0xff0000") : Color.decode("0x999999"));
            }
            if (focusEvent.getComponent() == packageField) {
                packageLabel.setForeground(isSelect ? Color.decode("0xff0000") : Color.decode("0x999999"));
            }
        }

        @Override
        public void focusGained(FocusEvent focusEvent) {
            select(focusEvent, true);
        }

        @Override
        public void focusLost(FocusEvent focusEvent) {
            select(focusEvent, false);
        }
    }

    public interface DialogCallBack {
        void ok(InputContentData data);
    }


}

File: src/main/java/MvvmDialog.kt
import Utils.activityToLayout
import java.awt.Color
import java.awt.event.*
import javax.swing.*
import javax.swing.event.ChangeEvent
import javax.swing.event.DocumentEvent
import javax.swing.event.DocumentListener

class MvvmDialog(packageName: String, var modeName: String, dialogOk: (InputContentData) -> Unit) : JDialog() {
    lateinit var contentPane: JPanel
    lateinit var buttonOK: JButton
    lateinit var buttonCancel: JButton
    lateinit var nameField: JTextField
    lateinit var descField: JTextField
    lateinit var activityNameField: JTextField
    lateinit var viewModelNameField: JTextField
    lateinit var layoutNameField: JTextField
    lateinit var packageField: JTextField
    lateinit var nameLabel: JLabel
    lateinit var descLabel: JLabel
    lateinit var activityNameLabel: JLabel
    lateinit var layoutNameLabel: JLabel
    lateinit var viewModelNameLabel: JLabel
    lateinit var packageLabel: JLabel
    lateinit var activityRadioButton: JRadioButton
    lateinit var fragmentRadioButton: JRadioButton
    lateinit var isListRadioButton: JRadioButton
    lateinit var isBindingRadioButton: JRadioButton
    private fun initView(packageName: String) {
        packageField.text = packageName
        //定义按钮组
        val group = ButtonGroup()
        group.add(activityRadioButton)
        group.add(fragmentRadioButton)
        activityRadioButton.addChangeListener({ changeEvent: ChangeEvent? -> onNameChange() })
        fragmentRadioButton.addChangeListener({ changeEvent: ChangeEvent? -> onNameChange() })

        //焦点改变
        nameField.addFocusListener(MyFocusListener())
        descField.addFocusListener(MyFocusListener())
        activityNameField.addFocusListener(MyFocusListener())
        viewModelNameField.addFocusListener(MyFocusListener())
        layoutNameField.addFocusListener(MyFocusListener())
        packageField.addFocusListener(MyFocusListener())
        nameField.document.addDocumentListener(object : DocumentListener {
            override fun insertUpdate(documentEvent: DocumentEvent) {
                onNameChange()
            }

            override fun removeUpdate(documentEvent: DocumentEvent) {
                onNameChange()
            }

            override fun changedUpdate(documentEvent: DocumentEvent) {
                onNameChange()
            }
        })
    }

    fun onNameChange() {
        val modeNameSux = if (modeName.isEmpty()) "" else modeName + "_"
        var nameText = nameField.text.trim { it <= ' ' }
        if (!nameText.isEmpty()) {
            if (nameText.lowercase().contains("activity")) {
                nameText = nameText.lowercase().replace("activity", "")
            }
            if (nameText.lowercase().contains("viewmodel")) {
                nameText = nameText.lowercase().replace("viewmodel", "")
            }
            viewModelNameField.text = nameText + "ViewModel"
            if (activityRadioButton.isSelected) {
                activityNameField.text = nameText + "Activity"
                layoutNameField.text = modeNameSux + "activity" + activityToLayout(nameText)
            } else {
                activityNameField.text = nameText + "Fragment"
                layoutNameField.text = modeNameSux + "fragment" + activityToLayout(nameText)
            }
        } else {
            activityNameField.text = ""
            viewModelNameField.text = ""
            layoutNameField.text = ""
        }
    }

    private fun createUIComponents() {
        // TODO: place custom component creation code here
    }

    private inner class MyFocusListener() : FocusListener {
        fun select(focusEvent: FocusEvent, isSelect: Boolean) {
            nameLabel.foreground = Color.decode("0x999999")
            descLabel.foreground = Color.decode("0x999999")
            activityNameLabel.foreground = Color.decode("0x999999")
            layoutNameLabel.foreground = Color.decode("0x999999")
            viewModelNameLabel.foreground = Color.decode("0x999999")
            packageLabel.foreground = Color.decode("0x999999")
            if (focusEvent.component === nameField) {
                nameLabel.foreground = if (isSelect) Color.decode("0xff0000") else Color.decode("0x999999")
            }
            if (focusEvent.component === descField) {
                descLabel.foreground = if (isSelect) Color.decode("0xff0000") else Color.decode("0x999999")
            }
            if (focusEvent.component === activityNameField) {
                activityNameLabel.foreground = if (isSelect) Color.decode("0xff0000") else Color.decode("0x999999")
            }
            if (focusEvent.component === layoutNameField) {
                layoutNameLabel.foreground = if (isSelect) Color.decode("0xff0000") else Color.decode("0x999999")
            }
            if (focusEvent.component === viewModelNameField) {
                viewModelNameLabel.foreground = if (isSelect) Color.decode("0xff0000") else Color.decode("0x999999")
            }
            if (focusEvent.component === packageField) {
                packageLabel.foreground = if (isSelect) Color.decode("0xff0000") else Color.decode("0x999999")
            }
        }

        override fun focusGained(focusEvent: FocusEvent) {
            select(focusEvent, true)
        }

        override fun focusLost(focusEvent: FocusEvent) {
            select(focusEvent, false)
        }
    }

    init {
        setContentPane(contentPane)
        isModal = true
        getRootPane().defaultButton = buttonOK
        initView(packageName)
        buttonOK.addActionListener(ActionListener {
            val data = InputContentData()
            data.isActivity = activityRadioButton.isSelected
            data.isList = isListRadioButton.isSelected
            data.isViewBinding = isBindingRadioButton.isSelected
            data.name = nameField.text.trim { it <= ' ' }
            data.desc = descField.text.trim { it <= ' ' }
            data.view = activityNameField.text.trim { it <= ' ' }
            data.viewMode = viewModelNameField.text.trim { it <= ' ' }
            data.layout = layoutNameField.text.trim { it <= ' ' }
            data.packge = packageField.text.trim { it <= ' ' }
            dialogOk.invoke(data)
            dispose()
        })
        buttonCancel.addActionListener(object : ActionListener {
            override fun actionPerformed(e: ActionEvent) {
                dispose()
            }
        })

        // call onCancel() when cross is clicked
        defaultCloseOperation = DO_NOTHING_ON_CLOSE
        addWindowListener(object : WindowAdapter() {
            override fun windowClosing(e: WindowEvent) {
                dispose()
            }
        })

        // call onCancel() on ESCAPE
        contentPane.registerKeyboardAction(object : ActionListener {
            override fun actionPerformed(e: ActionEvent) {
                dispose()
            }
        }, KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0), JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT)
    }
}